### PR TITLE
git: Fix add safe directory button doing nothing

### PR DESCRIPTION
**Pull request by cole-miller**
We had a nested spawn that was disguising the fact that the actual `git_config` task was being dropped immediately instead of detached. This only seems to cause problems in release builds due to timing issues.

Self-Review Checklist:

- [x] I've reviewed my own diff for quality, security, and reliability
- [x] Unsafe blocks (if any) have justifying comments
- [x] The content is consistent with the [UI/UX checklist](https://github.com/zed-industries/zed/blob/main/CONTRIBUTING.md#uiux-checklist)
- [x] Tests cover the new/changed behavior
- [x] Performance impact has been considered and is acceptable

Release Notes:

- Fixed a bug where clicking the button to trust a git repository would do nothing.

### CODE REVIEW

**Comment by probably-neb** [comment]
@zed-zippy approve

File: crates/git_ui/src/git_panel.rs [hunk 1]
```diff
@@ -3192,12 +3192,10 @@ impl GitPanel {
                 path_arg,
             ];
 
-            cx.spawn_in(window, async move |git_panel, cx| {
-                git_panel.update(cx, |git_panel, cx| {
-                    git_panel.project.read(cx).git_config(path, args, cx)
-                })
```
Comment (Anthony-Eid) [inline crates/git_ui/src/git_panel.rs:3198]: I wonder if dylint could've caught that we needed a .detach() after the .git_config here. Ok we immediately dropped this task without doing anything with it

@miguelraz

Comment (cole-miller) [reply crates/git_ui/src/git_panel.rs:3198]: Yes! We have must_use on `Task` but it doesn't penetrate generics :( :( :(
